src/backend/gl/src/lib.rs

```diff
@@ -53,10 +55,46 @@ pub type Texture        = gl::types::GLuint;
 pub type Sampler        = gl::types::GLuint;
 
 #[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
-pub struct Fence(gl::types::GLsync);
+struct RawFence(gl::types::GLsync);
 
-unsafe impl Send for Fence {}
-unsafe impl Sync for Fence {}
+impl RawFence {
+    fn wait(&self, gl: &gl::Gl) {
+        unsafe {
+            let timeout = 1_000_000_000_000;
+            gl.ClientWaitSync(self.0, gl::SYNC_FLUSH_COMMANDS_BIT, timeout);
```

Comment:
This can be called by multiple objects wanting to ensure they have exclusive CPU access. How much does this call costs ? The status of the fence could be cached to avoid calling this more than once.